File: OtherAlgorithmStudy/Hackerrank/Implementation/ElectronicsShop.cpp
//
//  ElectronicsShop.cpp
//  OtherAlgorithmStudy
//
//  Created by MinKyeongTae on 26/09/2019.
//  Copyright © 2019 Min Kyeong Tae. All rights reserved.
//

#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int getMoneySpent(vector<int> keyboards, vector<int> drives, int b) {
    sort(keyboards.begin(), keyboards.end());
    sort(drives.begin(), drives.end());
    int Ans = 0;
    for(int i=0; i<keyboards.size(); i++) {
        for(int j=0; j<drives.size(); j++) {
            if(keyboards[i]+drives[j]<=b) Ans = max(Ans, keyboards[i]+drives[j]);
            else break;
        }
    }
    return (Ans==0) ? -1 : Ans;
}

File: OtherAlgorithmStudy/Hackerrank/Implementation/ElectronicsShop.swift
//
//  ElectronicsShop.swift
//  OtherAlgorithmStudy
//
//  Created by MinKyeongTae on 26/09/2019.
//  Copyright © 2019 Min Kyeong Tae. All rights reserved.
//

import Foundation

func getMoneySpent(keyboards: [Int], drives: [Int], b: Int) -> Int {
    var Ans = 0
    for i in keyboards.sorted().indices {
        for j in drives.sorted().indices {
            if keyboards[i] + drives[j] <= b {
                let sum = keyboards[i] + drives[j]
                Ans = (Ans < sum) ? sum : Ans
            }
        }
    }
    return (Ans == 0) ? -1 : Ans
}
